### PR TITLE
NAS-121762 / 23.10 / Updates to Sort filter in new Apps

### PR DESCRIPTION
**Pull request by AlexKarpov98**
Check ticket description for more details.
Try different scenarios by changing filtering data :)

### CODE REVIEW

File: src/app/pages/apps/store/available-apps-store.service.ts [hunk 1]
```diff
@@ -393,4 +361,120 @@ export class AvailableAppsStore extends ComponentStore<AvailableAppsState> {
       }),
     );
   }
+
+  private sortAppsByName(filteredApps: AvailableApp[]): AppsByCategory[] {
+    const appsByCategory: AppsByCategory[] = [];
+
+    const alphabet = [
```
Comment (undsoft) [inline src/app/pages/apps/store/available-apps-store.service.ts:368]: Not really sustainable. There may be an app starting with a number or cyrillic letter. Need to look at existing letters in catalog apps.

File: src/app/pages/apps/services/applications.service.ts [hunk 2]
```diff
@@ -140,4 +143,47 @@ export class ApplicationsService {
   stopApplication(name: string): Observable<Job<ChartScaleResult>> {
     return this.ws.job('chart.release.scale', [name, { replica_count: 0 }]);
   }
+
+  convertDateToRelativeDate(date: Date): string {
+    const diff = Math.round(((new Date() as unknown as number) - (date as unknown as number)) / 1000);
```
Comment (undsoft) [inline src/app/pages/apps/services/applications.service.ts:148]: 1. You may be able to use `formatRelative` from `date-fns` here.
2. Seems to be impossible to test this right now, but it looks like granularity is too much. It should be something like `Last 24 hours`, `Last 3 days`, `Last week`, `Last month`, not `5 hours ago`, `5 days ago`.

**Review by undsoft — CHANGES_REQUESTED** [review]

File: src/app/pages/apps/store/available-apps-store.service.ts (hunk 1 above)
Comment (AlexKarpov98) [reply src/app/pages/apps/store/available-apps-store.service.ts:368]: @undsoft - you mean check all the apps names and get unique first letters?

File: src/app/pages/apps/services/applications.service.ts (hunk 2 above)
Comment (AlexKarpov98) [reply src/app/pages/apps/services/applications.service.ts:148]: @undsoft  - `date-fns` provides a strange solution with time, I was not able to find solution to fix it, 
If we can keep my method, I can remove unnecessary options and leave only few of them like:
`Last 24 hours, Last 3 days, Last week, Last month, Long time ago`

Screenshot 2023-05-03 at 17 45 17

Comment (undsoft) [reply src/app/pages/apps/services/applications.service.ts:148]: Okay, let's do that.

File: src/app/pages/apps/store/available-apps-store.service.ts (hunk 1 above)
Comment (undsoft) [reply src/app/pages/apps/store/available-apps-store.service.ts:368]: Yup.

**Review by undsoft — APPROVED** [review]
Looks good.

**Pull request merged**